File: Roadmap/13 - PRUEBAS UNITARIAS/javascript/AChapeton.js
// ! NOTA: El archivo utiliza la libreria Jest. Para utilizarlo, seguir los siguientes pasos:
// 1. Se debe de ejecutar el comando "npm i --save-dev jest"
// 2. En la lista de scripts dentro de package.json, incluir "test": "jest"
// 3. Cambiar el nombre del archivo a "AChapeton.test.js"

const sum = (a, b) => {
  return a + b
}

describe('Funcion Sum', () => {
  test('Debe de sumar dos numeros', () => {
    expect(sum(1, 2)).toBe(3)
  })
})

// DIFICULTAD EXTRA

const diccionario = {
  name: "Andres", 
  age: 26,
  birth_date: '02/06/1997',
  programming_languages: [
    "TypeScript", "JavaScript", "Python"
  ]
}

describe('Diccionario', () => {
  test('Deben de existir los 4 campos', () => {
    const keys = Object.keys(diccionario)
    expect(keys).toContain('name')
    expect(keys).toContain('age')
    expect(keys).toContain('birth_date')
    expect(keys).toContain('programming_languages')
  })

  test('Los datos deben de ser correctos', () => {
    const values = Object.values(diccionario)
    expect(values).toContain('Andres')
    expect(values).toContain(26)
    expect(values).toContain('02/06/1997')
    expect(values[3]).toEqual([
      "TypeScript", "JavaScript", "Python"
    ])
  })
})

File: Roadmap/13 - PRUEBAS UNITARIAS/python/AChapeton.py
import unittest

def sumar(num1, num2):
  return num1 + num2

class TestSum(unittest.TestCase):
  def test_1(self):
    resultado = sumar(1, 2)
    self.assertEqual(resultado, 3)

# DIFICULTAD EXTRA

user = {
    "name": "Chape",
    "age": 27,
    "birth_date": "1997-06-02",
    "programming_languages": ["js", "ts", "py"]
}

class TestDict(unittest.TestCase):
    def test_check_all_keys(self):
        self.assertIn("name", user)
        self.assertIn("age", user)
        self.assertIn("birth_date", user)
        self.assertIn("programming_languages", user)
    
    def test_check_values(self):
        self.assertEqual(user["name"], "Andres")
        self.assertEqual(user["age"], 27)
        self.assertEqual(user["birth_date"], "1997-06-02",)
        self.assertEqual(user["programming_languages"], ["js", "ts", "py"])


if __name__ == '__main__':
    unittest.main()

